### ISSUE TITLE
Rundeck 2.9 fails executions due to mixed up nodes

### ISSUE THREAD

**Issue by pbenas** [rundeck/rundeck#2756]
Issue type: Bug report

**My Rundeck detail**

* Rundeck version: 2.9.3
* install type:war
* OS Name/version: CentOS Linux release 7.3.1611
* DB Type/version: postgresql-9.2.18

**Expected Behavior**
When everything goes fine, rundeck performs 4 ssh connections to the target node:
1. scp
2. ssh - chmod
3. ssh execution
4. ssh - rm 

And the script was executed.

**Actual Behavior**
Random subset of target nodes fails with the `chmod` command reporting `No such file or directory`. Inspecting the `secure` log on the target instance, only one ssh command login was attempted by rundeck server, which must have been `ssh - chmod`. 

Which is even weirder is it seems rundeck mixes up different nodes (will attach a screenshot) such as:

1. scp to the correct node A
2. ssh to a node B (different node), where it cannot find the script for node A (see the screenshot).

The steps fails with the ENOENT.

**How to reproduce Behavior**
- Configure Rundeck to use script `script-copy` and `script-exec` as follows:
```
"service.FileCopier.default.provider": "script-copy",
"service.NodeExecutor.default.provider": "script-exec",
"plugin.script-copy.default.command": "/usr/bin/scp -q -o StrictHostKeyChecking=no ${file-copy.file} ${node.username}@${node.hostname}:${file-copy.destination}",
"plugin.script-copy.default.remote-filepath": "/tmp/${file-copy.filename}",
"plugin.script-exec.default.command": "/usr/bin/ssh -q -o StrictHostKeyChecking=no ${node.username}@${node.hostname} ${exec.command}"
```
- Execute a job containing a script on about hundread nodes.

**Comment by pbenas** [comment]
![screenshot](https://user-images.githubusercontent.com/10399279/30374906-d4cca970-9886-11e7-94bb-ef26a1517f8f.png)

**Comment by pbenas** [comment]
This seems to be a regression of Rundeck 2.9. Rundeck 2.8.4 is fine, this started since 2.9.0.

**Comment by geraldh** [comment]
Same problem here - if an attribute is missing from a node, the **missing attribute is taken from a previous node**.

Example - we are using rundeck for backups - attribute _services_ describes the services to stop/start during a backup run.

```
node1:
backupdirs: /etc:/var:/root:/srv
services: httpd:postgresql
```
```
node2:
backupdirs: /etc:/var:/root
```

If node2 is running after node1, node2 suddenly has a spurious attribute services with value "httpd:postgresql" - since rundeck 2.9.3

With rundeck 2.8.2 the value of the missing attribute services is "${node.services}" if we start our job with a parameter like: --services "${node.services}"

Also not really what you expect - I would say the correct value for missing node attributes should be the empty string "" .

Please fix - our backups are failing randomly at the moment.

**Comment by commarla** [comment]
Same thing here.

Running a job on 15 servers (with thread_count to 6). In the middle of `www-06` execution, rundeck switches to `www-04`

![screenshot 2017-09-20 17 11 10](https://user-images.githubusercontent.com/4509360/30651892-3fdfc922-9e27-11e7-9e55-7da58b2ffbf2.png)

**Comment by MarcFRICOU** [comment]
can't wait for a fix :pray: deployment are crazily long

**Comment by gschueler** [comment]
closed as dupe of #2766 which is fixed in 2.9.4
